### ISSUE TITLE
Version 12, simple mapping not working

### ISSUE THREAD

**Issue by Holt59** [tekezo/Karabiner-Elements#1372]
I have recently upgraded to v12, and I tried to create simple modifications:

![screen shot 2018-04-24 at 08 13 02](https://user-images.githubusercontent.com/2393288/39169279-827c9e30-4797-11e8-8ef4-1a812d105282.png)

And it does not work...

When I open the EventViewer, I see the events on these to keys in their original positions:

![screen shot 2018-04-24 at 08 12 48](https://user-images.githubusercontent.com/2393288/39169291-8ba346d0-4797-11e8-8d3d-33dafbdb5595.png)

I am on HighSierra 10.13.4, I tried restarting Karabiner, my Mac, applications I tested, nothing worked...

**Comment by tekezo** [comment]
Does another modification also not work?
For example, `non_us_backslash` -> `tab`.

**Comment by Holt59** [comment]
@tekezo This works except if one of the key is `non_us_backslash` or `grave_accent_and_tilde`.

**Comment by Holt59** [comment]
@tekezo No, without the simple modifications the behaviour is the same.

**Comment by tekezo** [comment]
Hmm, it's strange.

A last possibility is you are using other keyboard drivers which changes these keys before Karabiner-Elements.
Are you using another driver?

**Comment by justcoon** [comment]
hi, i got same problem, before v12, it worked, after upgrade to v12 it does not work

**Comment by Holt59** [comment]
@tekezo No, I am just using a QWERTY US-International layout on an AZERTY keyboard, but everything was working fine prior to v12.

**Comment by greelgorke** [comment]
@tekezo i'm using Karabiner also and it suddenly stoped working at all:
<img width="980" alt="remapping of modifiers" src="https://user-images.githubusercontent.com/2396468/39384579-99bfe1dc-4a6d-11e8-89d7-1361d979c614.png">

doesn't work anymore. Internal keyboard, nothing else changed.

**Comment by greelgorke** [comment]
this is the startup log:

`[2018-04-27 22:49:24.202] [info] [console_user_server] version 12.0.0
[2018-04-27 22:49:24.203] [info] [console_user_server] Check version...
[2018-04-27 22:49:24.221] [info] [console_user_server] Load karabiner.json...
[2018-04-27 22:49:24.224] [info] [console_user_server] core_configuration is updated.
[2018-04-27 22:49:24.225] [info] [console_user_server] Check for updates...
[2018-04-27 22:49:25.221] [info] [console_user_server] input_source_manager initialize
[2018-04-27 22:49:25.280] [info] [grabber] karabiner_console_user_server is connected (pid:2123)
[2018-04-27 22:49:25.280] [info] [console_user_server] receiver is initialized
[2018-04-27 22:49:25.280] [info] [console_user_server] Check version...
[2018-04-27 22:49:25.280] [info] [console_user_server] grabber_client_ is connected
[2018-04-27 22:49:25.281] [info] [grabber] event_tap_manager initialized
[2018-04-27 22:49:25.281] [info] [grabber] Load karabiner.json...
[2018-04-27 22:49:25.283] [info] [grabber] core_configuration is updated.
[2018-04-27 22:49:25.284] [info] [grabber] Apple Internal Keyboard / Trackpad (device_id:15) is ignored.
[2018-04-27 22:49:25.284] [info] [grabber] MX Master (device_id:16) is ignored.
[2018-04-27 22:49:25.284] [warning] [grabber] virtual_hid_device_client is not connected yet. Please wait for a while.
[2018-04-27 22:49:25.284] [info] [grabber] (vendor_id:0x5ac, product_id:0x8600) (device_id:13) is ignored.
[2018-04-27 22:49:25.288] [info] [console_user_server] Load karabiner.json...
[2018-04-27 22:49:25.291] [info] [console_user_server] core_configuration is updated.
[2018-04-27 22:49:25.370] [info] [console_user_server] frontmost_application_observer initialize
[2018-04-27 22:49:25.371] [info] [console_user_server] input_source_observer initialize
[2018-04-27 22:49:28.438] [info] [console_user_server] system_preferences is updated.
[2018-04-27 22:49:28.439] [info] [grabber] system_preferences_updated`

**Comment by tekezo** [comment]
@Holt59 
Is the keyboard type configuration correct?
https://pqrs.org/osx/karabiner/document.html#configuration-keyboard-type

@greelgorke 
Please confirm the target device is checked on Karabiner-Elements > Preferences > Devices tab.

**Comment by greelgorke** [comment]
@tekezo it is checked

**Comment by greelgorke** [comment]
@tekezo i'd like to provide any info you need. i got so used to my setup, i barely can use other keyboards for serious stuff :D

**Comment by dmitriyK1** [comment]
I also have this problem with recent version of Karabiner

**Comment by zuyu** [comment]
I am also using v12 on 10.13.4, and a simple modification that maps `caps lock` to `escape` does not work after applying the latest security update 2018-001.

On the other hand, I found that `system preferences`->`keyboard` now optionally supports above key modification. So I guess something has changed regarding the key mappings.

Finally I would like to thank Karabiner-Elements team for this great software!

**Comment by squashmcdermott** [comment]
version 12 stopped working for me also.

**Comment by Holt59** [comment]
@tekezo I checked the keyboard type and I went from ISO to ANSI. The keys are already in the correct position, so this is ok for me, but Karabiner still does not map events from/to these keys.

**Comment by greelgorke** [comment]
@zuyu this capslock to escape mapping is available since the osx version that came alongside the mac book pro with the touchbar. so it like at least a year available.

But i agree that last patch from( what was that?) last week? two weeks ago? made karabiner not working

**Comment by zuyu** [comment]
Hi @greelgorke, [it](https://github.com/tekezo/Karabiner-Elements/issues/1372#issuecomment-385317901) was released on Apr 24.

**Comment by zuyu** [comment]
Hi @greelgorke, there is no issues w/ the the security patch, and `Karabiner-Elements` works for me.

tl;dr
I just applied the latest security patch on another `Macbook Pro`, and the capslock remapping using `Karabiner-Elements` works fine on that.

Also, after reinstall `Karabiner-Elements` on the previous `Macbook Pro`, it also works now!

**Comment by dmitriyK1** [comment]
@zuyu 
I'm having trouble swapping non_us_backslash(§) and grave_accent_and_tilde(`), does it work for you know?

**Comment by greelgorke** [comment]
@zuyu @tekezo The tip with uninstall, restart, install fixed the issue for me. After that, the VirtualHIDKeyboard appeared in the Keyboard Pane In System Settings. It seems like the update routine from 11 to 12 doesn't work properly.

**Comment by elitan** [comment]
I had the exact same problem swapping grave_accent_and_tilde(`)  to non_us_backslash(§). 

I uninstalled Karabiner and installed it again and now it is working again.

**Comment by stavros-papadopoulos** [comment]
I have the exact same problem after updating to version 12.

**Comment by dmitriyK1** [comment]
I found how to fix it in my case - I opened ```~/.config/karabiner/karabiner.json``` and search for all occurences of "grave_accent_and_tilde" and deleted rules containing this (there were 2 of them but only one shown in karabiner UI)
Looks like some old rules left in my config and prevented to properly change this mapping
Hope this'll be helpful to someone

**Comment by bejibx** [comment]
Same problem, swapping § to ` is not working. Reinstall did not help. @dmitriyK1 solution did not help either.

**Comment by kannanmavila** [comment]
I had the same issue and turned out I had to go to `identify the keyboard` in `Preferences > Keyboard` and run the wizard. Fixed the issue for me.

The option doesn't appear on `Preferences > Keyboard` anymore hence no screenshot. Sorry.
